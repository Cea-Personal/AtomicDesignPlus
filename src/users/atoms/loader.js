import React, { useEffect } from "react";
import { TimelineMax } from "gsap";
import styled from "styled-components";
import Dumbell from "../../assets/logo.png";
import PropTypes from "prop-types";
const Div = styled.div`
  display: flex;
  width: 70px;
  height: 70px;
  justify-content: center;
  align-items: center;
`;
const Img = styled.img`
  width: 100%;
  height: 100%;
  margin: 0;
`;
let loaderDiv;
export const tween =(value, action) =>
  new TimelineMax({ paused: true}).fromTo(
    value,
    0.25,
    {
      opacity: 1
    },
    { opacity: 0.3, delay: 0.25, onComplete: () => action }
  );

const Loader = props => {
  useEffect(() => {
    props.loading &&
      tween(loaderDiv, tween(loaderDiv).reverse())
        .repeat(-1)
        .play();
  }, []);

  return (
    <Div>
      <Img src={Dumbell} ref={div => (loaderDiv = div)} />
    </Div>
  );
};
export default Loader;

Loader.propTypes = {
  loading: PropTypes.bool
};
Loader.defaultProps = {
  loading: false
};

