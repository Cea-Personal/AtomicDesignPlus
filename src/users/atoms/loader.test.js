import React from "react";
import { render, cleanup, fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Loader from './loader';
import {tween} from './loader'
jest.useFakeTimers();
afterEach(cleanup);
it('loader should display', ()=> {
    const {getByRole, container, asFragment} = render(
        <Loader />
      );
      const ImageContent =getByRole('img')
      expect(container).toBeDefined()
      expect(ImageContent).toBeDefined();
      expect(asFragment()).toMatchSnapshot()
})
it('loader should animate if loading is true', () => {
    const {getByRole} = render(
        <Loader loading={true} />
      );
      const ImageContent = getByRole('img')
      let style = window.getComputedStyle(ImageContent)
      expect(style.opacity).toBe('1')
      fireEvent.load(ImageContent,tween(ImageContent).vars.paused = false)
     
     style = window.getComputedStyle(ImageContent)
     console.log(style)
    //  expect(tweened).toHaveBeenLastCalledWith(expect.any(Function), 1000);
})
it('should reverse on complete',()=>{

})
it('should not display if loading is false',()=>{

})
