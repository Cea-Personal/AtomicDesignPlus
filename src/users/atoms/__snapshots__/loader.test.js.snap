// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should display 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa fdDfIe"
  >
    <img
      class="sc-bwzfXH eHxbrp"
      src="logo.png"
    />
  </div>
</DocumentFragment>
`;
